app/models/exam_registration/authorization_criterion.rb

```diff
@@ -0,0 +1,61 @@
+class ExamRegistration::AuthorizationCriterion
+  attr_reader :value
+
+  def initialize(value)
+    @value = value
+  end
+
+  def type
+    self.class.name.demodulize.underscore
+  end
+
+  def as_json
+    { type: type, value: value }
+  end
+
+  def ensure_valid!
+    raise "Invalid criterion value #{value} for #{type}" unless valid?
+  end
+
+  def process_request!(authorization_request)
+    authorization_request.update! status: authorization_status_for(authorization_request.user)
+  end
+
+  def authorization_status_for(user)
+    enabled_for?(user) ? :approved : :rejected
```

Comment:
I would change `enabled_for?` for something like `meets_authorization_criteria?`

Reply:
Done!